### DISCUSSION TITLE
Doubt about license

### DISCUSSION THREAD

**Discussion by esuig** [artem-ogre/CDT#107] [General]
Hello, 

hope it is not a trivial question. I tried to study the license but I'm not sure about an aspect. Can I use this library in a closed source commercial application without redistributing the code? I would not modify the library, just use it as it is now and it would be used to perform constrained triangulations.

Many thanks

**Comment by artem-ogre** [comment]
Hi, @esuig,

Short answer is yes, it is possible in this case. You only need to inform the users that you use CDT and where they can obtain its source code from.

> You must inform the recipients where they can get the source for the MPLed code in the executable program or library you are distributing

Check Q8 of the official [MPL-2.0 FAQ](https://www.mozilla.org/en-US/MPL/2.0/FAQ/) and Section 3.2 of the [MPL-2.0 license](https://www.mozilla.org/en-US/MPL/2.0/) for the details.

**Comment by artem-ogre** [comment]
I am obviously curious about CDT usages: may I ask what application is it? :)

**Comment by esuig** [comment]
Hello, 
I'm still evaluating, application does not exist yet, however it should be related to pathfinding.

**Comment by esuig** [comment]
So, to inform the user I could for example include a file in the documentation of the software, or among the installation files, is this correct?

**Comment by artem-ogre** [comment]
For example, that's how Google Maps for iOS do it: they have "Open-source licenses" entry in settings "About, terms and privacy".
And that's how they mention Eigen (which is MPL-2.0 too):
![image](https://user-images.githubusercontent.com/7488293/198974986-62b3bdbe-aa66-4ea2-9ba2-ab1c0a617aa4.png)

So you could use:
```
This software is based part on CDT (C++ library for constrained Delaunay triangulation):
Copyright © 2019 Leica Geosystems Technology AB
Copyright © The CDT Contributors
Licensed under the MPL-2.0 license.
https://github.com/artem-ogre/CDT
${INSERT_FULL_MPL_2.0_TEXT}
```
**Update:** edited the template sightly

**Comment by esuig** [comment]
Thank you very much
